### PR TITLE
Added variation to NPC chat snippets

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Content "Adds new snippets to NPC chat activity."```

#### Purpose of change
As recommended in #27570 , I've added a lot more variety to the text generated when "chitchatting" with an NPC, with some conditionals for various things the NPC may comment on.

#### Describe the solution
Mostly flavour text, but the NPC will notice the season, the time of day, if they are infected, if your doing a mission for them, or if you have beer in your inventory, and add a comment or two.

#### Describe alternatives you've considered
N/A

#### Additional context
N/A

### CODE REVIEW

File: data/json/npcs/talk_tags.json [hunk 1]
```diff
@@ -73,6 +73,30 @@
             "Water... Is there an oasis nearby?",
             "Did you know that lack of water kills faster than lack of rest?"
         ]
+    },{
+        "type" : "snippet",
+        "category" : "<noticedbooze>",
+        "text" : [
+            "Yeah sure, can't help but notice you got beer with you! let's crack a cold one and chat, <name_g>, how goes it?",
```
Comment (Night-Pryanik) [inline data/json/npcs/talk_tags.json]:
```suggestion
            "Yeah sure, can't help but notice you got beer with you! Let's crack a cold one and chat, <name_g>, how goes it?",
```

File: data/json/npcs/talk_tags.json [hunk 2]
```diff
@@ -73,6 +73,30 @@
...
+            "Oh definitely, how about one of those beers I see on you?, what's up anyway?",
```
Comment (Night-Pryanik) [inline data/json/npcs/talk_tags.json]:
```suggestion
            "Oh definitely, how about one of those beers I see on you? What's up anyway?",
```

File: data/json/npcs/talk_tags.json [hunk 3]
```diff
@@ -73,6 +73,30 @@
...
+            "Yeah you share those beers I see you hoarding and then we chat all you like!... only joking, what's up <name_g>?"
```
Comment (Night-Pryanik) [inline data/json/npcs/talk_tags.json]:
```suggestion
            "Yeah you share those beers I see you hoarding and then we chat all you like! Only joking, what's up <name_g>?"
```

File: data/json/npcs/talk_tags.json [hunk 4]
```diff
@@ -73,6 +73,30 @@
...
+        ]
+    },{
+        "type" : "snippet",
+        "category" : "<neutralchitchat>",
+        "text" : [
+            "Hey, sure thing, <name_g>, I need a break anyway, how are you?",
+            "Yeah OK, <name_g>, how's it going?",
+            "Sure, let's shoot the shit!, you OK?",
```
Comment (Night-Pryanik) [inline data/json/npcs/talk_tags.json]:
```suggestion
            "Sure, let's shoot the shit! You OK?",
```

File: data/json/npcs/talk_tags.json [hunk 5]
```diff
@@ -73,6 +73,30 @@
...
+            "Why not!, how you doing?",
```
Comment (Night-Pryanik) [inline data/json/npcs/talk_tags.json]:
```suggestion
            "Why not? How you doing?",
```

File: data/json/npcs/talk_tags.json [hunk 6]
```diff
@@ -73,6 +73,30 @@
...
+            "I'm OK with that, what's up?",
+            "I can spare a few minutes, how's things?",
+            "Sure thing <name_g>, you good?",
+            "Alright, you got something to get off your chest?",
+            "Always ready for a good chat! but why, you OK?",
```
Comment (Night-Pryanik) [inline data/json/npcs/talk_tags.json]:
```suggestion
            "Always ready for a good chat! But why, you OK?",
```

File: data/json/npcs/talk_tags.json [hunk 7]
```diff
@@ -73,6 +73,30 @@
...
+            "OK <name_g>, we should get to know each other, how are you coping?",
+            "Definitely, I'm game. how you holding up?"
```
Comment (Night-Pryanik) [inline data/json/npcs/talk_tags.json]:
```suggestion
            "Definitely, I'm game. How you holding up?"
```

File: data/json/npcs/TALK_COMMON_OTHER.json [hunk 8]
```diff
@@ -27,7 +27,42 @@
...
+        "is_night": "Man it's dark out isn't it? what's up?"
+      },
+      {
+        "npc_has_effect": "infected",
+        "yes": "Well, I'm feeling pretty sick... are you doing OK though?",
+        "no": "<neutralchitchat>"
+      },
+      {
+        "u_has_mission": true,
+        "many": "Definitely, by the way, thanks for helping me so much with my tasks! Anyway, you coping OK, <name_g>? ",
```
Comment (ZhilkinSerg) [inline data/json/npcs/TALK_COMMON_OTHER.json]:
```suggestion
        "many": "Definitely, by the way, thanks for helping me so much with my tasks!  Anyway, you coping OK, <name_g>? ",
```

File: data/json/npcs/talk_tags.json [hunk 9]
```diff
@@ -73,6 +73,30 @@
             "Water... Is there an oasis nearby?",
             "Did you know that lack of water kills faster than lack of rest?"
         ]
+    },{
+        "type" : "snippet",
+        "category" : "<noticedbooze>",
+        "text" : [
+            "Yeah sure, can't help but notice you got beer with you! Let's crack a cold one and chat, <name_g>, how goes it?",
+            "Oh definitely, how about one of those beers I see on you? What's up anyway?",
```
Comment (ZhilkinSerg) [inline data/json/npcs/talk_tags.json]:
```suggestion
            "Oh definitely, how about one of those beers I see on you?  What's up anyway?",
```

File: data/json/npcs/talk_tags.json [hunk 10]
```diff
@@ -73,6 +73,30 @@
...
+            "Yeah you share those beers I see you hoarding and then we chat all you like! Only joking, what's up <name_g>?"
```
Comment (ZhilkinSerg) [inline data/json/npcs/talk_tags.json]:
```suggestion
            "Yeah you share those beers I see you hoarding and then we chat all you like!  Only joking, what's up <name_g>?"
```

File: data/json/npcs/talk_tags.json [hunk 11]
```diff
@@ -73,6 +73,30 @@
...
+        ]
+    },{
+        "type" : "snippet",
+        "category" : "<neutralchitchat>",
+        "text" : [
+            "Hey, sure thing, <name_g>, I need a break anyway, how are you?",
+            "Yeah OK, <name_g>, how's it going?",
+            "Sure, let's shoot the shit! You OK?",
```
Comment (ZhilkinSerg) [inline data/json/npcs/talk_tags.json]:
```suggestion
            "Sure, let's shoot the shit!  You OK?",
```

File: data/json/npcs/talk_tags.json [hunk 12]
```diff
@@ -73,6 +73,30 @@
...
+            "Why not? How you doing?",
```
Comment (ZhilkinSerg) [inline data/json/npcs/talk_tags.json]:
```suggestion
            "Why not?  How you doing?",
```

File: data/json/npcs/talk_tags.json [hunk 13]
```diff
@@ -73,6 +73,30 @@
...
+            "I'm OK with that, what's up?",
+            "I can spare a few minutes, how's things?",
+            "Sure thing <name_g>, you good?",
+            "Alright, you got something to get off your chest?",
+            "Always ready for a good chat! But why, you OK?",
```
Comment (ZhilkinSerg) [inline data/json/npcs/talk_tags.json]:
```suggestion
            "Always ready for a good chat!  But why, you OK?",
```

File: data/json/npcs/talk_tags.json [hunk 14]
```diff
@@ -73,6 +73,30 @@
...
+            "OK <name_g>, we should get to know each other, how are you coping?",
+            "Definitely, I'm game. How you holding up?"
```
Comment (ZhilkinSerg) [inline data/json/npcs/talk_tags.json]:
```suggestion
            "Definitely, I'm game.  How you holding up?"
```

File: data/json/npcs/talk_tags.json [hunk 15]
```diff
@@ -73,6 +73,30 @@
             "Water... Is there an oasis nearby?",
             "Did you know that lack of water kills faster than lack of rest?"
         ]
+    },{
+        "type" : "snippet",
+        "category" : "<noticedbooze>",
+        "text" : [
+            "Yeah sure, can't help but notice you got beer with you! Let's crack a cold one and chat, <name_g>, how goes it?",
```
[the remark this answers is not in the document]
Comment (davidpwbrown) [reply data/json/npcs/talk_tags.json]: Why two spaces?

[the remark this answers is not in the document]
Comment (mlangsdorf) [reply data/json/npcs/talk_tags.json]: Manual of style specifies two spaces after punctuation.

**Pull request merged**